api/api.go

```diff
@@ -46,3 +47,37 @@ type SyncerPeer struct {
 type SyncerConnectRequest struct {
 	NetAddress string `json:"netAddress"`
 }
+
+// A ContractBaseRequest contains the information of the host we are dialing
+// for contract related requests.
+type ContractBaseRequest struct {
+	NetAddress string          `json:"netAddress"`
+	TheirKey   types.PublicKey `json:"theirKey"`
+}
```

Comment:
I'd just call this `ScanRequest`, and inline the fields in `FormRequest` and `RenewRequest`